xgcm/grid.py

```diff
@@ -1306,14 +1306,20 @@ def set_metrics(self, key, value, overwrite=False):
     def _get_dims_from_axis(self, da, axis):
         dim = []
         for ax in axis:
-            all_dim = self.axes[ax].coords.values()
-            matching_dim = [di for di in all_dim if di in da.dims]
-            if len(matching_dim) == 1:
-                dim.append(matching_dim[0])
+            set_ax = frozenset(_maybe_promote_str_to_list(ax))
+            if set_ax.issubset(set(self.axes.keys())):
+                all_dim = self.axes[ax].coords.values()
+                matching_dim = [di for di in all_dim if di in da.dims]
+                if len(matching_dim) == 1:
+                    dim.append(matching_dim[0])
+                else:
+                    raise ValueError(
+                        "Did not find single matching dimension %s from %s corresponding to axis %s. Got (%s)"
+                        % (da.dims, da.name, ax, matching_dim)
```

Comment:
Lets f-string these. They're easier to read.

Feel free to ignore this suggestion